src/newgrf_engine.cpp

```diff
@@ -702,6 +707,12 @@ static uint32 VehicleGetVariable(Vehicle *v, const VehicleScopeResolver *object,
 			return ret;
 		}
 
+		case 0x63: { // Poweredness relative to arbitrary railtype
+			if (v->type != VEH_TRAIN) return 0;
+			RailType rt = GetTileRailType(v->tile);
+			return HasPowerOnRail(object->ro.grffile->railtype_map[parameter], rt) ? 1 : 0;
```

Comment:
Railtype map has as length of RAILTYPE_END (64 currently), while a buggy could pass any parameter. I.e. missing limit check.